### PR TITLE
Knative manifest now uses valid port names

### LINKED ISSUE #10260: quarkus.kubernetes.annotations gets not applied to ingress
**Describe the bug**
I'm using the kubernetes extension to deploy to kubernetes. 
I'm setting `expose` to `true` to get an ingress resource generated (`target/kubernetes/kubernetes.xml`).
My annotation `quarkus.kubernetes.annotations."kubernetes.io/ingress.class"=addon-http-application-routing` gets not applied to the generated ingress but to all other resources.

**Expected behavior**
Labels and annotations gets applied to ingress too.

**Actual behavior**
Can not apply annotations via application.properties to generated ingress resource.

**To Reproduce**
1. Bootstrap a new Quarkus project using the kubernetes extension
2. Add the following application.properties

```
quarkus.kubernetes.deployment-target=kubernetes
quarkus.kubernetes.expose=true
quarkus.kubernetes.annotations."kubernetes.io/ingress.class"=addon-http-application-routing
```

3. Call maven `mvn package`
4. Open `target/kubernetes/kubernetes.xml` file to see the generated resources.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 1.5.2.Final

### LINKED ISSUE #10332: Generated knative.yaml does not work on Openshift Serverless
**Describe the bug**
When using the `quarkus-kubernetes` extension and `quarkus.kubernetes.deployment-target=knative`, the generated manifest cannot be applied to an Openshift Cluster with Openshift Serverless installed.

**Expected behavior**
The auto-generated manifests work OOTB on Openshift Serverless

**Actual behavior**
When applying the manifests as produced by the Kubernetes extension, we get:

```
Error from server (BadRequest): error when creating "target/kubernetes/knative.json": admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: Port name http is not allowed: spec.template.spec.containers[0].ports
Name must be empty, or one of: 'h2c', 'http1'
```

However, if we remove the `ports` section inside the `container`, the manifest gets applied properly.

### PR DESCRIPTION
**Pull request by iocanel**
Resolves #10332 
Resolves #10260

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by iocanel** [comment]
@geoand: since the bump fixes more than a single issue. I added two more test cases just to verify that we also address #10260. You might want to have a second look.

File: integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithApplicationPropertiesTest.java [hunk 1]
```diff
@@ -66,6 +67,14 @@ public void assertGeneratedResources() throws IOException {
             });
         });
 
-        assertThat(openshiftList).filteredOn(h -> "Route".equals(h.getKind())).hasSize(1);
+        assertThat(openshiftList).filteredOn(i -> "Route".equals(i.getKind())).hasOnlyOneElementSatisfying(i -> {
+            assertThat(i).isInstanceOfSatisfying(Route.class, in -> {
+                //Check that lables are also applied to Routes (#10260)
+                assertThat(i.getMetadata()).satisfies(m -> {
+                    assertThat(m.getName()).isEqualTo("test-it");
+                    assertThat(m.getLabels()).contains(entry("foo", "bar"));
```
Comment (geoand) [inline integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithApplicationPropertiesTest.java:75]: Can we also add a check for annotations?

**Comment by gsmet** [comment]
All the JVM tests have failed so there's probably something wrong.

File: integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithApplicationPropertiesTest.java (hunk 1 above)
Comment (geoand) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithApplicationPropertiesTest.java:75]: @iocanel can you please add a check for annotations as well?

Comment (iocanel) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithApplicationPropertiesTest.java:75]: Sure

**Comment by geoand** [comment]
@gsmet I am going to mark this for backport since it's purely bug fixes

**Pull request merged**